seqeralabs/seqerakit
# Draft: Add feature to emit YAMLs of existing entities

Partially closes #83 to add a `seqerakit dump` feature to emit YAML definitions for right now only CEs and Pipelines. Also improves CLI option readability and usability.

To test:
Install this branch locally:
```bash
git checkout dump_yamls
pip install .
```
Run the dump command:
```bash
seqerakit dump -w <any_workspace> -p <optional_prefix>
```
For example:
```bash
seqerakit dump -w seqeralabs/showcase -p showcase_dump
```
Additionally, to see the improvements to CLI arg parser:
```bash
seqerakit -h
```

TODO:
- Improve support in YAML parsing for lists 
- Potentially add support for Datasets as well
- Refactor the `DumpYaml` class (kind of mess rn)
- Add logging
